### PR TITLE
Dynamic empty text

### PR DESCRIPTION
**Pull request by peterjcoffman**
Hoist P/R Checklist
-------------------

Review and check off the below. Items that do not apply can also be checked off to indicate they
have been considered. If unclear if a step is relevant, please leave unchecked and note in comments.

- [x] Up to date with `develop` branch as of last change.
- [x] Ready for review (or open as a draft PR / add `wip` label).
- [x] Added CHANGELOG entry (or N/A)
- [x] Reviewed for breaking changes (add `breaking-change` label + CHANGELOG if so, or N/A)
- [x] Updated doc comments / prop-types (or N/A)
- [ ] Reviewed and tested on Mobile (or N/A)
- [x] Created Toolbox branch / PR (link provided here, or N/A)
https://github.com/xh/toolbox/tree/dynamicEmptyText
Not suggesting we pull these changes in, but might be useful to see.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: cmp/grid/Grid.js [hunk 1]
```diff
@@ -216,7 +216,11 @@ class LocalModel {
             rowDeselection: true,
             getRowHeight: () => this.rowHeight,
             getRowClass: ({data}) => model.rowClassFn ? model.rowClassFn(data) : null,
-            overlayNoRowsTemplate: model.emptyText || '<span></span>',
+            noRowsOverlayComponentFramework: hoistCmp({
+                render() {
+                    return model.emptyText;
+                }
+            }),
```
Comment (amcclain) [inline cmp/grid/Grid.js]: Here I would suggest collapsing this to one line with the more minimal syntax `hoistCmp(() => model.emptyText)`

**Comment by peterjcoffman** [comment]
> @lbwexler what do you think about the use of `@bindable` here w/o an explicit setter? I'm on the fence re. adding a setter - more boilerplate code, but easier to discover given current (lack of) tooling around those annotations and IDE/doc hints.

Why doesn't bindable define a 'proper' ES6-style setter?

**Comment by lbwexler** [comment]
Looking at this with Peter now.  We discussed the fact that we use setXXX() and at this point are unlikely to be able to change that.

**Comment by lbwexler** [comment]
Regarding Anselm's question -- would be happy to use an explicit setXXX method rather than bindable in this case, both for the new option and the existing showSummary().

On a related note,  groupBy should be @observable.ref.

**Comment by peterjcoffman** [comment]
@lbwexler I fixed those few things.

**Pull request merged**